R/shared-methods.R

```diff
@@ -0,0 +1,145 @@
+#' Check country and regions are avaliable and set up country class
+#'
+#' @description Check data for the requested country and region
+#' is avaliable and return an initialised region class for that country.
+#' @inheritParams get_regional_data
+#' @return The target countries specific object if avaliable, e.g. [Italy()]
+#' @rdname download_data
+check_country_avaliable <- function(country = character(), level = level,
```

Comment:
this recursive definition doesn't make sense? the args need to have default values

Reply:
```r
r$> check_country_avaliable(country = "Italy")                                                 
Error in stopifnot(is.character(level)) : 
  promise already under evaluation: recursive default argument reference or earlier problems?
```

Reply:
Agreed, I've set these value to actual defaults which removes the error when called.